test/functional/http/post/1.second.secret.js

```diff
@@ -0,0 +1,438 @@
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using invalid second passphrase should fail', function () {
+				transaction = node.lisk.transaction.createTransaction(node.eAccount.address, 1, account.password, 'invalid password');
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Failed to verify second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using correct second passphrase should be ok', function () {
+				transaction = node.lisk.transaction.createTransaction(node.eAccount.address, 1, account.password, account.secondPassword);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using correct empty second passphrase should be ok', function () {
+				transaction = node.lisk.transaction.createTransaction(node.eAccount.address, 1, accountEmptySecondPassword.password, accountEmptySecondPassword.secondPassword);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactionsEnforcement.push(transaction);
+				});
+			});
+		});
+
+		describe('type 1 - second secret', function () {
+
+			it('setting second signature twice on the same account should be not ok', function () {
+				transaction = node.lisk.signature.createSignature(account.password, node.randomPassword());
+				var secondKeys = node.lisk.crypto.getKeys(account.secondPassword);
+				node.lisk.crypto.secondSign(transaction, secondKeys);
+				transaction.id = node.lisk.crypto.getId(transaction);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.not.be.ok;
+					node.expect(res).to.have.property('message').to.equal('Missing sender second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+		});
+
+		describe('type 2 - registering delegate', function () {
+
+			it('using no second passphrase should fail', function () {
+				transaction = node.lisk.delegate.createDelegate(account.password, account.username);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Missing sender second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using invalid second passphrase should fail', function () {
```

Comment:
using second passphrase that does not match registered secondPublicKey should fail